extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/WebSocketIdentityUpdateProvider.java

```diff
@@ -0,0 +1,73 @@
+package io.quarkus.oidc.runtime;
+
+import static io.quarkus.oidc.common.runtime.OidcConstants.INTROSPECTION_TOKEN_SUB;
+import static io.quarkus.vertx.http.runtime.security.HttpSecurityUtils.getRoutingContextAttribute;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import io.quarkus.oidc.AccessTokenCredential;
+import io.quarkus.oidc.OidcTenantConfig;
+import io.quarkus.oidc.TokenIntrospection;
+import io.quarkus.security.AuthenticationFailedException;
+import io.quarkus.security.identity.AuthenticationRequestContext;
+import io.quarkus.security.identity.IdentityProvider;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.spi.runtime.BlockingSecurityExecutor;
+import io.quarkus.websockets.next.runtime.spi.telemetry.WebSocketIdentityUpdateRequest;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+@ApplicationScoped
+public class WebSocketIdentityUpdateProvider implements IdentityProvider<WebSocketIdentityUpdateRequest> {
+
+    @Inject
+    DefaultTenantConfigResolver resolver;
+
+    @Inject
+    BlockingSecurityExecutor blockingExecutor;
+
+    WebSocketIdentityUpdateProvider() {
+    }
+
+    @Override
+    public Class<WebSocketIdentityUpdateRequest> getRequestType() {
+        return WebSocketIdentityUpdateRequest.class;
+    }
+
+    @Override
+    public Uni<SecurityIdentity> authenticate(WebSocketIdentityUpdateRequest request,
+            AuthenticationRequestContext authenticationRequestContext) {
+        return authenticate(request.getCredential().getToken(), getRoutingContextAttribute(request))
+                .onItem().transformToUni(newIdentity -> {
+                    TokenIntrospection introspection = OidcUtils.getAttribute(newIdentity, OidcUtils.INTROSPECTION_ATTRIBUTE);
```

Comment:
@michalvavrik I think we should start with enforcing that the `sub` is available, either via `TokenIntrospection` or JWT. You can check if it is JWT checking if `newIdentity.getPrincipal()` is an instance of `JsonWebToken` - if it is, the `sub` claim must be available, if not, it must be in the introspection response.
That may not work with OAuth2 providers, but I believe we should start in a strict mode.

Reply:
The `sub` claim check is now enforced.